Sources/FuturedKit/NavigationLinks.swift

```diff
@@ -0,0 +1,61 @@
+import SwiftUI
+///
+/// ViewModifier and protocol for creating and handling NavigationLinks
+///
+/// ```swift
+/// enum HomeLinks: NavigationLinks {
+///     case detail, settings, profile
+/// }
+///
+/// struct HomeScene: some View {
+///     @State navigationState: HomeLinks?
+///
+///     var body: some View {
+///        homeViewContent.pushNavigation {
+///             HomeLinks.detail.link(to: DetailScene(), selection: $navigationState)
+///             HomeLinks.settings.link(to: SettingsScene(), selection: $navigationState)
+///             HomeLinks.profile.link(to: ProfileScene(), selection: $navigationState)
+///     }
+///
+///     func showDetail() {
+///         navigationState = .detail
+///     }
+/// }
+/// ```
+
+/// Protocol, by which you should specify and create links to different destinations
```

Comment:
The documentation won't be easily discoverable, since it is not connected to the type definition and separated by an empty line.